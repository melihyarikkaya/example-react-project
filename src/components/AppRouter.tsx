import React, { Component } from 'react';
import { BrowserRouter as Router, Route, Link } from 'react-router-dom';

//Components
import Home from './Home';
import Loading from '../containers/LoadingContainer';

export default class AppRouter extends Component {
  render() {
    return (
      <Router>
        <div>
          <ul>
            <li>
              <Link to="/">Home</Link>
            </li>
            <li>
              <Link to="/loading">Loading</Link>
            </li>
          </ul>
          <Route exact path="/" component={Home} />
          <Route path="/loading" component={Loading} />
        </div>
      </Router>
    );
  }
}
