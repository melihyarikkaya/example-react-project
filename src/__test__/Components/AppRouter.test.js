import React from 'react';
import { shallow } from 'enzyme';

import AppRouter from '../../components/AppRouter';

describe('AppRouter Component', () => {
  describe('Snapshot', () => {
    let wrapper = null;
    beforeEach(() => {
      wrapper = shallow(<AppRouter />);
    });
    afterEach(() => {
      wrapper = null;
    });
    it('should match last snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
    it('should have two <li> tag in <ul> tag', () => {
      expect(wrapper.find('ul>li')).toHaveLength(2);
    });
  });
});
