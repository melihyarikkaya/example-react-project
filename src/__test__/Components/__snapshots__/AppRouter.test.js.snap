// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppRouter Component Snapshot should match last snapshot 1`] = `
<BrowserRouter>
  <div>
    <ul>
      <li>
        <Link
          replace={false}
          to="/"
        >
          Home
        </Link>
      </li>
      <li>
        <Link
          replace={false}
          to="/loading"
        >
          Loading
        </Link>
      </li>
    </ul>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      path="/loading"
    />
  </div>
</BrowserRouter>
`;
